### PR TITLE
Move from peep to pip8

### PR DESCRIPTION
**Pull request by pmclanahan**

### CODE REVIEW

**Comment by pmclanahan** [comment]
@jgmize here's the just pip8 PR. I'll deploy to a demo.

**Comment by pmclanahan** [comment]
After a few tweaks I've got a successful [deployment to demo3](https://www-demo3.allizom.org/en-US/)

**Comment by jgmize** [comment]
Awesome, lgtm so far; I'm just doing a final test to make sure that "git push deis" works and confirm there aren't any overlayfs-specific issues.

**Comment by jgmize** [comment]
> Awesome, lgtm so far; I'm just doing a final test to make sure that "git push deis" works and confirm there aren't any overlayfs-specific issues.

Unfortunately there are:

```
$ git push deis
Counting objects: 82109, done.
Delta compression using up to 4 threads.
Compressing objects: 100% (28006/28006), done.
Writing objects: 100% (82109/82109), 332.36 MiB | 1.97 MiB/s, done.
Total 82109 (delta 50997), reused 78893 (delta 48366)

-----> Building Docker image
remote: Sending build context to Docker daemon 154.1 MB
Step 0 : FROM mozorg/bedrock_base
 ---> e6bc78dffe4f
Step 1 : COPY . /app
 ---> cc580b90e818
Removing intermediate container cc07f12521a1
Step 2 : RUN ./bin/pipstrap.py
 ---> Running in c6a4a26113be
  Exception:
  Traceback (most recent call last):
    File "/usr/local/lib/python2.7/dist-packages/pip/basecommand.py", line 246, in main
      pip_version_check(session)
    File "/usr/local/lib/python2.7/dist-packages/pip/commands/install.py", line 352, in run
      if options.target_dir:
    File "/usr/local/lib/python2.7/dist-packages/pip/req/req_set.py", line 704, in install
      def install(self, install_options, global_options=(), *args, **kwargs):
    File "/usr/local/lib/python2.7/dist-packages/pip/req/req_install.py", line 743, in commit_uninstall
      os.path.join(bin_dir, name) + '.exe'
    File "/usr/local/lib/python2.7/dist-packages/pip/req/req_uninstall.py", line 153, in commit
      self.file = pth_file
    File "/usr/local/lib/python2.7/dist-packages/pip/_vendor/retrying.py", line 49, in wrapped_f
      return Retrying(*dargs, **dkw).call(f, *args, **kw)
    File "/usr/local/lib/python2.7/dist-packages/pip/_vendor/retrying.py", line 212, in call
      raise attempt.get()
    File "/usr/local/lib/python2.7/dist-packages/pip/_vendor/retrying.py", line 247, in get
      six.reraise(self.value[0], self.value[1], self.value[2])
    File "/usr/local/lib/python2.7/dist-packages/pip/_vendor/retrying.py", line 200, in call
      attempt = Attempt(fn(*args, **kwargs), attempt_number, False)
    File "/usr/local/lib/python2.7/dist-packages/pip/utils/__init__.py", line 67, in rmtree
      SUPPORTED_EXTENSIONS += XZ_EXTENSIONS
    File "/usr/lib/python2.7/shutil.py", line 247, in rmtree
      rmtree(fullname, ignore_errors, onerror)
    File "/usr/lib/python2.7/shutil.py", line 247, in rmtree
      rmtree(fullname, ignore_errors, onerror)
    File "/usr/lib/python2.7/shutil.py", line 252, in rmtree
      onerror(os.remove, fullname, sys.exc_info())
    File "/usr/local/lib/python2.7/dist-packages/pip/utils/__init__.py", line 75, in rmtree_errorhandler
      except ImportError:
  OSError: [Errno 2] No such file or directory: '/tmp/pip-MVl3fZ-uninstall/usr/bin/pip'

Traceback (most recent call last):
  File "./bin/pipstrap.py", line 118, in <module>
    exit(main())
  File "./bin/pipstrap.py", line 105, in main
    shell=True)
  File "/usr/lib/python2.7/subprocess.py", line 573, in check_output
    raise CalledProcessError(retcode, cmd, output=output)
subprocess.CalledProcessError: Command 'pip install --no-index --no-deps -U /tmp/pipstrap-f4bDUM/pip-8.0.2.tar.gz /tmp/pipstrap-f4bDUM/setuptools-19.4.tar.gz /tmp/pipstrap-f4bDUM/wheel-0.26.0.tar.gz' returned non-zero exit status 2
The command '/bin/sh -c ./bin/pipstrap.py' returned a non-zero code: 1
```

Looks like we hit https://github.com/docker/docker/issues/12327 as I [feared](https://github.com/mozilla/bedrock/pull/3906#issuecomment-189134907). However, I think we can work around it again; I'll do some experimentation.

**Comment by jgmize** [comment]
The circleci demo deployments eliminate the need for the "git push deis" workflow, so the overlayfs bug is no longer an issue. Once this is rebased I think we're good to merge.

r+wc, thanks @pmclanahan

**Comment by pmclanahan** [comment]
Agreed. And while we're not sure yet if we're keeping Circle, I'm pretty confident that we should keep the CI built demo app workflow no matter the CI. I'll get this rebased shortly.

**Comment by pmclanahan** [comment]
Added a little sugar that avoids calling `pipstrap.py` every time since CircleCI caches the virtualenv but pipstrap doesn't check to see if pip 8 is already installed.

**Comment by pmclanahan** [comment]
I think this is ready now.

File: Dockerfile [hunk 1]
```diff
@@ -5,9 +5,10 @@ RUN apt-get install -y --no-install-recommends npm
 COPY . /app
 
 RUN npm install --production
-RUN ./bin/peep.py install --no-cache-dir -r requirements/dev.txt
-RUN ./bin/peep.py install --no-cache-dir -r requirements/prod.txt
-RUN ./bin/peep.py install --no-cache-dir -r requirements/docker.txt
+RUN ./bin/pipstrap.py
```
Comment (jgmize) [inline Dockerfile]: I don't see why this would be necessary, given that pipstrap was run in the base dockerfile.

**Comment by jgmize** [comment]
r+wc, nice work!

File: Dockerfile (hunk 1 above)
Comment (pmclanahan) [inline Dockerfile]: That's only true after this is merged, so it's not been run there yet. We can remove this later though.

Comment (jgmize) [inline Dockerfile]: Ah, I see. Thanks.

**Pull request merged**